### PR TITLE
Create method `TSDataset.flatten` 

### LINKED ISSUE #226: Make `TSDataset.to_flatten` static method
### 🚀 Feature Request

Make `TSDataset.to_pandas` method static

### Motivation

We need this method to work with pd.DataFrames in ETNA format, 
moreover, now `TSDataset.to_dataset` is a static method but `TSDataset.to_pandas` is not

### Proposal

make static

### Test cases

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### Checklist

- [X] Added feature request
- [X] Added motivation
- [X] Added proposal

### PR DESCRIPTION
**Pull request by alex-hse-repository**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna-ts/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [ ] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna-ts/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Proposed Changes

Create method `TSDataset.flatten` 
## Closing issues

closes #226

### CODE REVIEW

**Review by julia-shenshina — COMMENTED** [review]
Looks 👍🏼 
could you please add test that checks smth like
```
assert (df == TSDataset.to_flatten(TSDataset.to_dataset(df))).all().all()
```

File: etna/datasets/tsdataset.py [hunk 1]
```diff
@@ -342,6 +342,59 @@ def plot(self, n_segments: int = 10, column: str = "target", segments: Optional[
...
+        ... )
+        >>> df.head(5)
+            timestamp    segment  target
+        0  2021-06-01  segment_0    1.00
+        1  2021-06-02  segment_0    1.00
+        2  2021-06-03  segment_0    1.00
+        3  2021-06-04  segment_0    1.00
+        4  2021-06-05  segment_0    1.00
+        >>> df_ts_format = TSDataset.to_dataset(df)
+        >>> ts = TSDataset(df_ts_format, "D")
```
Comment (julia-shenshina) [inline etna/datasets/tsdataset.py:374]: actually you don't need to create `ts` here, you can just run
```
df_ts_format = TSDataset.to_dataset(df)
TSDataset.to_flatten(df_ts_format).head(5)
```
🤔

File: etna/datasets/tsdataset.py [hunk 2]
```diff
@@ -373,12 +426,12 @@ def to_pandas(self, flatten: bool = False) -> pd.DataFrame:
         >>> df_ts_format = TSDataset.to_dataset(df)
         >>> ts = TSDataset(df_ts_format, "D")
         >>> ts.to_pandas(True).head(5)
-        feature  timestamp  target    segment
-        0       2021-06-01    1.00  segment_0
-        1       2021-06-02    1.00  segment_0
-        2       2021-06-03    1.00  segment_0
-        3       2021-06-04    1.00  segment_0
-        4       2021-06-05    1.00  segment_0
+           timestamp  target    segment
```
Comment (julia-shenshina) [inline etna/datasets/tsdataset.py:429]: does it look good in docs?

Comment (alex-hse-repository) [reply etna/datasets/tsdataset.py:429]: Looks ok

**Review by julia-shenshina — DISMISSED** [review]

File: tests/test_datasets/test_dataset.py [hunk 3]
```diff
@@ -384,3 +384,13 @@ def test_right_format_sorting():
     inv_df = tsd.to_pandas(flatten=True)
     pd.testing.assert_series_equal(df["reg_1"], inv_df["reg_1"])
     pd.testing.assert_series_equal(df["reg_2"], inv_df["reg_2"])
+'''
```
Comment (julia-shenshina) [inline tests/test_datasets/test_dataset.py:387]:
```suggestion

```

File: tests/test_datasets/test_dataset.py [hunk 4]
```diff
@@ -52,6 +51,7 @@ def df_and_regressors() -> Tuple[pd.DataFrame, pd.DataFrame]:
     return df, df_exog
 
 
+'''
```
Comment (julia-shenshina) [inline tests/test_datasets/test_dataset.py:54]:
```suggestion

```

File: tests/test_datasets/test_dataset.py (hunk 3 above)
Comment (alex-hse-repository) [inline tests/test_datasets/test_dataset.py:387]:
```suggestion

```

File: tests/test_datasets/test_dataset.py (hunk 4 above)
Comment (alex-hse-repository) [inline tests/test_datasets/test_dataset.py:54]:
```suggestion

```

**Review by julia-shenshina — DISMISSED** [review]

**Review by julia-shenshina — APPROVED** [review]

**Pull request merged**